### ISSUE TITLE
Generate Open API spec for Zaproxy API

### ISSUE THREAD

**Issue by omerlh** [zaproxy/zaproxy#3690]
It would be great to have Open API for the Zaproxy API as it make a lot of things (for example, client generation, API discovery)  pretty simple. The biggest issue here is the response type - currently, all the existing SDKs use a generic response type, and I think it is a bit useless to have Open API without the actual response type. By looking at the code, it seems pretty hard to support this and might require pretty big refactor - all `ApiImplementor` register only the parameters, not the response type.

**Comment by psiinon** [comment]
There is another (non trivial) option - creating a v2 API defined by Open API that really is a REST one.
We will need to maintain the current one for the foreseeable future, but we can support both, eg via urls like https://localhost:8080/v2/... for the new one.
Not something to be rushed into, but it would allow us to significantly refactor the API and bring it up to modern standards.
Thoughts?

**Comment by omerlh** [comment]
Yeah, it is also might be a good time to use proper REST framework like Spring. Not sure about Spring, as I am not familiar with Java and REAS API development, but we should choose one and use it. 
I think it will be easy to start with the alerts API, think it is the most useful now, and if it will work we can work on converting the others.

**Comment by psiinon** [comment]
Definitely agree with using a proper framework. The 2 APIs should still map to the same underlying code (ie what we are using now).
And yes, we should be able to come up with a core set of key endpoints (spider, scan, alerts) and then build it out from there.
Maybe one for 2.8.0?

**Comment by omerlh** [comment]
Yeah, I would like to help - but I could not do it alone, the task is pretty big.

**Comment by psiinon** [comment]
I think this is one that we can definitely distribute between a few of us :)
@thc202 @kingthorin what do you think? Is this something we should be looking at for 2.8.0?
If so we can start investigating & documenting suitable frameworks and approaches.
